fleetspeak/src/server/mysql/mysql.go

```diff
@@ -69,6 +72,33 @@ func (d *Datastore) runInTx(ctx context.Context, readOnly bool, f func(*sql.Tx)
 	return tx.Commit()
 }
 
+// runInTx runs f in a transaction. If the transaction returns a retriable MySQL
+// error, the transaction will be retried (and f will be run again).
+func (d *Datastore) runInTx(ctx context.Context, readOnly bool, f func(*sql.Tx) error) error {
+	var err error
+L:
+	for i := 0; i < maxRetries; i++ {
+		err = d.runOnce(ctx, readOnly, f)
+		e, ok := err.(*mysql.MySQLError)
+		if !ok {
+			return err
+		}
+		switch e.Number {
+		case 1213, 1205, 1637: // Deadlock, lock timeout, concurrent transactions
```

Comment:
What do you think about making those numbers constants and getting rid of the comment?

Reply:
At the top of the file? If they'd be reused somehow it would make sense, but as it is, it would feel like indirection for the sake of indirection. Here you have everything you need to compare behavior to the MySQL error code documentation.

Reply:
Done.